examples/core/gated_gcn/main.py

```diff
@@ -0,0 +1,124 @@
+import argparse
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from model import GatedGCN
+from ogb.graphproppred import collate_dgl, DglGraphPropPredDataset, Evaluator
+from torch.utils.data import DataLoader
+
+
+def train(model, device, data_loader, opt, loss_fn):
+    model.train()
+    train_loss = []
+
+    for g, labels in data_loader:
+        g = g.to(device)
+        labels = labels.to(torch.float32).to(device)
+        logits = model(g, g.edata["feat"], g.ndata["feat"])
+        loss = loss_fn(logits, labels)
+        opt.zero_grad()
+        loss.backward()
+        opt.step()
+        train_loss.append(loss.item())
+
+    return sum(train_loss) / len(train_loss)
+
+
+@torch.no_grad()
+def evaluate(model, device, data_loader, evaluator):
+    model.eval()
+    y_true, y_pred = [], []
+
+    for g, labels in data_loader:
+        g = g.to(device)
+        logits = model(g, g.edata["feat"], g.ndata["feat"])
+        y_true.append(labels.detach().cpu())
+        y_pred.append(logits.detach().cpu())
+
+    y_true = torch.cat(y_true, dim=0).numpy()
+    y_pred = torch.cat(y_pred, dim=0).numpy()
+
+    return evaluator.eval({"y_true": y_true, "y_pred": y_pred})["rocauc"]
+
+
+def main(args):
+    device = "cuda:0" if torch.cuda.is_available() else "cpu"
+
+    # Load ogb dataset & evaluator.
+    dataset = DglGraphPropPredDataset(name="ogbg-molhiv")
+    evaluator = Evaluator(name="ogbg-molhiv")
+
+    n_classes = dataset.num_tasks
+
+    split_idx = dataset.get_idx_split()
+    train_loader = DataLoader(
```

Comment:
See if you can use `GraphDataLoader` with the default `collate_fn` to be consistent with other DGL examples.